google/cloud/aiplatform/matching_engine/matching_engine_index_endpoint.py

```diff
@@ -1266,14 +1269,16 @@ def deploy_index(
             "Deploy index", "index_endpoint", self.__class__, deploy_lro
         )
 
-        deploy_lro.result(timeout=None)
+        # if sync then wait for any LRO to complete
+        if sync:
```

Comment:
The explicit check of `sync` should not be necessary with the `@base.optional_sync()` decorator. Please see https://github.com/googleapis/python-aiplatform/blob/ead5c931e708281720ac6ef4e22c40312e2f7cae/google/cloud/aiplatform/base.py#L810